x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/AuthenticationServiceTests.java

```diff
@@ -1511,6 +1511,7 @@ public void testRealmAuthenticateTerminateAuthenticationProcessWithException() {
         verifyNoMoreInteractions(operatorPrivilegesService);
     }
 
+    @AwaitsFix(bugUrl = "TODO")
```

Comment:
We deliberately call `onFailure(null)` here:

https://github.com/elastic/elasticsearch/blob/bb44f7f30392f4a4f9ae418757092aa653898e62/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RealmsAuthenticator.java#L186

I am suppressing this test temporarily to see if there are any other places that use a null exception, but will revert this change before marking this as ready-to-review.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/94328 to address this; apart from that this is ready to review

Reply:
https://github.com/elastic/elasticsearch/pull/94328 is merged so I've reinstated the null check